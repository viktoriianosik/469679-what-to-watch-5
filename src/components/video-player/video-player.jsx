import React, {Fragment, PureComponent} from "react";
import PropTypes from "prop-types";

class VideoPlayer extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {isPlaying, muted, onTimeUpdate} = this.props;

    return (
      <Fragment>
        <video autoPlay={isPlaying} ref={this.props.innerRef} className="player__video" muted={muted} onTimeUpdate={onTimeUpdate}></video>
      </Fragment>
    );
  }
}

VideoPlayer.propTypes = {
  isPlaying: PropTypes.bool.isRequired,
  muted: PropTypes.bool.isRequired,
  videoRef: PropTypes.string,
  onTimeUpdate: PropTypes.func.isRequired,
  innerRef: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({current: PropTypes.instanceOf(Element)})
  ]),
};

export default React.forwardRef((props, ref) => <VideoPlayer
  innerRef={ref} {...props}
/>);
