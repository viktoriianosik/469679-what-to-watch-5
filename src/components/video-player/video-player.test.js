import React from "react";
import renderer from "react-test-renderer";
import {VideoPlayer} from "./video-player";

it(`Render component VideoPlayer correctly`, () => {
  const tree = renderer.create(
      <VideoPlayer
        isPlaying={true}
        muted={false}
        onTimeUpdate={() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
