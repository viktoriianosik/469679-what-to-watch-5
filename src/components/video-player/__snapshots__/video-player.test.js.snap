// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component VideoPlayer correctly 1`] = `
<video
  autoPlay={true}
  className="player__video"
  muted={false}
  onTimeUpdate={[Function]}
/>
`;
